compiler/src/dotty/tools/dotc/quoted/QuoteContextImpl.scala

```diff
@@ -45,9 +45,9 @@ object QuoteContextImpl {
 
 }
 
-class QuoteContextImpl private (ctx: Context) extends QuoteContext:
+class QuoteContextImpl private (ctx: Context) extends QuoteContext, scala.internal.quoted.CompilerInterface:
 
-  object reflect extends scala.tasty.Reflection, scala.internal.tasty.CompilerInterface:
+  object reflect extends scala.tasty.Reflection:
```

Comment:
There are three concepts here:

- CompilerInterface
- Reflection
- QuoteContext

The responsibility for QuoteContext is relatively clear, it's the facade and it mainly holds API for `Type[T]` and `Expr[T]`.

It seems there is some overlapping of responsibilities between `Reflection` and `CompilerInterface`:

- `CompilerInterface` is a contract with the compiler, not the API for meta-programmers
- `Reflection` is both the contract with the compiler, and the API for meta-programmers

To continue the refactoring in #9818, it would be nice to make the responsibility of each interface clear and disjoint.

Reply:
I would say that this PR also removed the overlapping between `Reflection` and `CompilerInterface` as now the `CompilerInterface` only cares about `Expr` and `Type`, but nothing within `Reflection`. 

I will still try to make the internal interaction with `CompilerInterface` simpler.

Reply:
> I would say that this PR also removed the overlapping between `Reflection` and `CompilerInterface` as now the `CompilerInterface` only cares about `Expr` and `Type`, but nothing within `Reflection`.
> 

So rename `CompilerInterface` to `QuoteInterface` might make things more clear?

Reply:
That would make it clearer.